### PR TITLE
MIDIs are now either 'Meme' or 'Atmospheric', players can toggle each option

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Updated savefile number from 19 to 20. Meme and atmospheric preferences are enabled by default.

Admin sounds now need a selection between 'Meme' or 'Atmospheric' type. Ideally, this would let players decide if they want to listen to hijack or first drop songs without needing to listen to GOOD HITS or whatnot.

I am uncertain about the savefile bit of code. I don't fully understand it.

As stated I don't care about GBP, so if the tags are teechnicallly incorrect go ahead and change them or whatever.

# Explain why it's good for the game

> Admin sounds now need a selection between 'Meme' or 'Atmospheric' type. Ideally, this would let players decide if they want to listen to hijack or first drop songs without needing to listen to GOOD HITS or whatnot.

As it says. Lots of people hate the memes and just want to listen to the cool atmosphere. This is of course dependant on staff selecting the right option, which is sometimes up to opinion, but I fully trust staff will be able to handle this subjective affair correctly.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
refactor: Updated savefile number from 18 to 19. Meme and atmospheric preferences are enabled by default.
admin: Admin sounds now need a selection between 'Meme' or 'Atmospheric' type. Ideally, this would let players decide if they want to listen to hijack or first drop songs without needing to listen to GOOD HITS or whatnot.
/:cl:

### CODE REVIEW

File: code/__DEFINES/__game.dm [hunk 1]
```diff
@@ -90,12 +90,14 @@
...
-#define SOUND_INTERNET	16
-#define SOUND_REBOOT	32
+#define SOUND_ADMINHELP 1
+#define SOUND_MIDI 2
+#define SOUND_AMBIENCE 4
+#define SOUND_LOBBY	8
+#define SOUND_INTERNET 16
+#define SOUND_REBOOT 32
+#define SOUND_ADMIN_MEME 64
+#define SOUND_ADMIN_ATMOSPHERIC	128
```
Comment (stanalbatross) [inline code/__DEFINES/__game.dm:100]: why mess up the numbers... also shouldn't this use flags ie 1<<0 1<<1 1<<2 etc

File: code/modules/client/preferences_toggles.dm [hunk 2]
```diff
@@ -396,6 +397,15 @@
...
 
+/client/proc/toggle_admin_sound_types()
+	var/result = tgui_alert(src, "Which sound type do you want to toggle?", "Toggle MIDI/Internet sound type to play", list("Meme", "Atmospheric"))
+	if(result == "Meme")
+		prefs.toggles_sound ^= SOUND_ADMIN_MEME
+		to_chat(src, SPAN_NOTICE("You will [prefs.toggles_sound & SOUND_ADMIN_MEME ? "now" : "no longer"] hear meme admin sounds."))
+	if(result == "Atmospheric")
+		prefs.toggles_sound ^= SOUND_ADMIN_ATMOSPHERIC
+		to_chat(src, SPAN_NOTICE("You will [prefs.toggles_sound & SOUND_ADMIN_ATMOSPHERIC ? "now" : "no longer"] hear atmospheric admin sounds."))
+
```
Comment (stanalbatross) [inline code/modules/client/preferences_toggles.dm:408]: make the list strings meme [enabled] etc so people can see what they have turned on and off

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/__DEFINES/__game.dm (hunk 1 above)
Comment (carlarctg) [reply code/__DEFINES/__game.dm:100]: i recall walter saying using tabs is bad and we should just do a single space to separate instead

but yes ok

File: code/modules/client/preferences_toggles.dm (hunk 2 above)
Comment (carlarctg) [reply code/modules/client/preferences_toggles.dm]: done

File: code/modules/admin/verbs/playsound.dm [hunk 3]
```diff
@@ -74,11 +80,11 @@
 
 	for(var/i in targets)
 		var/mob/M = i
-		var/client/C = M?.client
-		if(C?.prefs.toggles_sound & SOUND_INTERNET)
-			C?.tgui_panel?.play_music(web_sound_url, music_extra_data)
+		var/client/client = M?.client
+		if(client?.prefs.toggles_sound & SOUND_INTERNET && client?.prefs.toggles_sound & sound_type)
```
Comment (stanalbatross) [inline code/modules/admin/verbs/playsound.dm:84]:
```suggestion
		if((client?.prefs.toggles_sound & SOUND_INTERNET) && (client?.prefs.toggles_sound & sound_type))
```

File: code/modules/admin/verbs/playsound.dm [hunk 4]
```diff
@@ -148,13 +160,13 @@
 
 	for(var/items in targets)
 		var/mob/Mob = items
-		var/client/Client = Mob?.client
-		if(Client?.prefs.toggles_sound & SOUND_MIDI)
-			admin_sound.volume = vol * Client?.admin_music_volume
+		var/client/client = Mob?.client
+		if(client?.prefs.toggles_sound & SOUND_INTERNET && client?.prefs.toggles_sound & sound_type)
```
Comment (stanalbatross) [inline code/modules/admin/verbs/playsound.dm:164]:
```suggestion
		if((client?.prefs.toggles_sound & SOUND_INTERNET) && (client?.prefs.toggles_sound & sound_type))
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Pull request merged**